### PR TITLE
Hashing nodes

### PR DESCRIPTION
**Pull request by clrosenbluth**
-removed parent in tree nodes
-added hashmap for boards/nodes to avoid repetition
-added helper method in board class to get equivalent boards

### CODE REVIEW

File: src/main/java/touro/peg/TriangleBoard.java [hunk 1]
```diff
@@ -54,6 +55,22 @@ public TriangleBoard triangleBoardCopy() {
         return triangleBoardCopy;
     }
 
+    public ArrayList<TriangleBoard> getEquivalentBoards() {
```
Comment (andrewoid) [inline src/main/java/touro/peg/TriangleBoard.java:65]: Let's not worry about this for now.

**Review by andrewoid — COMMENTED** [review]
I notice you didn't add any tests.
How do you know this works?

**Comment by clrosenbluth** [comment]
While writing the tests it became very clear that this code needs an overridden hashcode method. Unfortunately I'm not sure how to do that in a way that generates the same hashcode for equivalent boards. Hopefully we can discuss this in class this week.

**Comment by andrewoid** [comment]
> While writing the tests it became very clear that this code needs an overridden hashcode method. Unfortunately I'm not sure how to do that in a way that generates the same hashcode for equivalent boards. Hopefully we can discuss this in class this week.

No.
IntelliJ will create equals and hashcode methods for you.

**Comment by andrewoid** [comment]
> I'm not sure how to do that in a way that generates the same hashcode for equivalent boards. 

Don't worry about equivalent boards.

**Review by andrewoid — APPROVED** [review]

**Comment by clrosenbluth** [comment]
Is there anything else that I should do before this can be merged?

**Pull request merged**